client/src/context/useTour.tsx

```diff
@@ -0,0 +1,47 @@
+import { useContext, useState, createContext, PropsWithChildren, useEffect } from 'react';
+import { ReactourStep } from 'reactour';
+import ITourContext from '../interface/TourContext';
+
+export const TourContext = createContext<ITourContext>({} as ITourContext);
+
+interface TourProviderProps {
+  steps: ReactourStep[];
+}
+
+export const TourProvider = ({ steps, children }: PropsWithChildren<TourProviderProps>): JSX.Element => {
+  const [isTourMode, setIsTourMode] = useState<boolean>(false);
+  const [hasSeenTour, setHasSeenTour] = useState<boolean>(false);
+  const [tourSteps] = useState<ReactourStep[]>(steps || []);
+
+  useEffect(() => {
+    const hasSeenTour = localStorage.getItem('hasSeenTour');
+    if (!hasSeenTour) {
+      setIsTourMode(true);
+    }
+  });
```

Comment:
I forgot to add in the empty dependency array so it runs on just render as opposed to right now which is running on render then on every update.